public-packages/schemaStream/src/utils/streaming-json-parser.ts

```diff
@@ -233,6 +238,14 @@ export class SchemaStream {
 
     parser.onToken = this.handleToken.bind(this)
     parser.onValue = () => void 0
+    parser.onEnd = () => {
```

Comment:
instead of passing a conditional callback that depends on another flag - would it be easier for us to just allow an onComplete here and just by default pass through the safeparse results always?

feels like making sure that isStrictSchmea is set on the instance AND havaing to pass the callback might be a bit confusing/unintuitive

vs say just adding

.parse({ onComplete })

and we conditional add the onEnd wiht a safeParse inside of it? We could also jsut pasas through the finala result and any other meta we have for free

```
  const stream = parser.parse({
    onComplete:({ isValid, errors = [], data }) => {
      errors = zodError.errors
    }
  })
  ```

Reply:
yeah I dig that, that way you can do whatever you want

Reply:
love it, updated